irods/password_obfuscation.py

```diff
@@ -56,7 +56,11 @@ def decode(s, uid=None):
 
     #The uid is used as a salt.
     if uid is None:
-        uid = os.getuid()
+        try:
+            uid = os.getuid()
+        except AttributeError:
+            # Spoof UID for Non-POSIX
+            uid = sum((ord(char) for char in os.getlogin()))
```

Comment:
If we were ever interested in a slightly improved cryptographic strength, we could always use something like `hash(os.getlogin( ))` or even a `int(hashlib.sha256(blah).hexdigest( ),16)`, rather of the sum of the character ord-values.  But hey,  this gets the job done : )

Reply:
In the end, it's just salt for local use.  It need only be consistent.  I don't know what extra strength is really required.

Reply:
@jpmcfarland Yes, that is definitely enough for the purpose at hand.

Reply:
Only one machine so far.  I have asked others if they can test using an editable pip install.  Not too many of us have access to unmanaged Windows machines.